tsm-client: 8.1.25.0 -> 8.1.27.0, drop old migration code

Yarny0 opened the pull request:
*   `tsm-client` module: drop migration code that [got added in 2023 for freeform settings](https://github.com/NixOS/nixpkgs/pull/253428)
*   ~~`tsm-client-gui` vm test: [migrate to `runTest`](https://github.com/NixOS/nixpkgs/issues/386873)~~
    Edit 2025-05-29: Meanwhile [done in another pull request](https://github.com/NixOS/nixpkgs/pull/410569) and [already merged](https://github.com/NixOS/nixpkgs/pull/410569#event-17830837270), so the corresponding commit is now dropped from the pull request at hand.
*   Edit 2025-03-31: `tsm-client` package: update to newest version 8.1.26.0
*   Edit 2025-06-21: `tsm-client` package: update to newest version 8.1.27.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

`nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"` tries to rebuild 3000+ package and was therefore skipped.

Edit 2025-03-31 & Edit 2025-06-21: More:

- [x] Both test derivations (cli and gui) build/pass.
- [x] Successfully tested with a real tsm-server: uploaded some files for backup without errors/problems.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Yarny0 committed:
nixos/tsm-client: drop migration code from 2023 freeform update

This manually reverts
20a9a21b246ca684e4e5a45e543a8a53be46e1d2.
Automatic revert is not possible due to nixfmt in
4f0dadbf38ee4cf4cc38cbc232b7708fddf965bc.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5289

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5323

Yarny0 committed:
tsm-client: 8.1.25.0 -> 8.1.26.0

"What's new" page:
https://www.ibm.com/docs/en/storage-protect/8.1.26?topic=new-backup-archive-client-updates

Other resources are hidden behind a login portal.

Yarny0 commented:
Pushed a rebase due to https://github.com/NixOS/nixpkgs/pull/380990

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5396

lluchs reviewed (approved):
Package builds and the test runs correctly on my machine.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5439

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5472

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5500

Yarny0 commented:
Branch updated and rebased to consider https://github.com/NixOS/nixpkgs/pull/410569 (which re-did the original intent of the pull-request at hand): The vmtest commit is no longer needed and dropped.

Yarny0 committed:
tsm-client: 8.1.26.0 -> 8.1.27.0

"What's new" page:
https://www.ibm.com/docs/en/storage-protect/8.1.27?topic=new-backup-archive-client-updates

Other resources are hidden behind a login portal.

doronbehar commented:
I'm not familiar with the changes in the 1st commit, but it doesn't look too bad, so I'll take the risk and merge this :).
